src/server/read_dir.rs

```diff
@@ -59,7 +59,7 @@ impl ReadDir {
     }
 }
 
-pub fn read_dir<D: AsRawFd>(dir: &mut D, offset: libc::c_long) -> Result<ReadDir> {
+pub fn read_dir<D: AsRawFd>(dir: &D, offset: libc::c_long) -> Result<ReadDir> {
```

Comment:
If `read_dir()` only requires `&impl AsRawFd` after this change, then we can probably also change the call sites that currently pass a `&mut File`:

```
    let mut dir = open_fid(proc, parent, P9_DIRECTORY)?; // <-- this doesn't need to be mut
    let mut dirents = read_dir(&mut dir, 0)?; // <-- this can pass &dir instead of &mut dir
```

```
        let dir = fid.file.as_mut().ok_or_else(ebadf)?;  // <--- this can now be as_ref()
        let mut dirents = read_dir(dir, readdir.offset as libc::c_long)?;
```